tests/utils.go

```diff
@@ -2084,12 +2084,14 @@ func GetGuestAgentUserData() string {
 	ipv6UserDataString := ""
 	if netutils.IsIPv6String(dnsServerIP) {
 		ipv6UserDataString = fmt.Sprintf(`sudo ip -6 addr add fd10:0:2::2/120 dev eth0
-                             sudo ip -6 route add default via fd10:0:2::1 src fd10:0:2::2
+                             for i in {1..20}; do sudo ip -6 route add default via fd10:0:2::1 src fd10:0:2::2 && break || sleep 0.1; done
```

Comment:
Do you mind adding a comment explaining any random reader what is this for? Yeah, they can read git history, but the info there can get lost after the function is moved.

Reply:
Comments tend to rot, developers should track down the git history for reasoning and intent.
The history cannot be lost, it may just be harder to track.

Based on the comment pattern, we will see more stories than code. For this specific script, one can also be interest to know why NM is disabled, why IPv6 is added here, why the route has a source address, etc.. this is impractical IMO.

Reply:
I think that if there is a code that cannot be understood from the codebase, knowledge of the language and understanding of the topic ... should be documented. But it is debatable and your point is fair.

Reply:
configureIPv6OnVMI this function also have same logic

Reply:
https://github.com/kubevirt/kubevirt/blob/master/tests/utils.go#L2842

> configureIPv6OnVMI this function also have same logic, in case it should be updated please

configureIPv6OnVMI already has a sleep [1]. Here it was missed.

Anyway, I would change the way we do the sleep in both of the places to be the same.

[1] https://github.com/kubevirt/kubevirt/blob/master/tests/utils.go#L2842

Reply:
Yes, it needs to be the same... but I think the current code in general is bad, so we will try to work on it as part of a refactoring effort.

Reply:
I think it is unrelated. You're fixing here something that was already done in 'configureIPv6OnVMI' but was missed here. The fix should be the same.
If we had a way to reuse the same code here and in 'configureIPv6OnVMI' it would be the best. But I don't have an idea how. And if you're going to refactor it anyway, maybe it doesn't worth the effort.

Reply:
For the refactor,
Keep in mind please that the two places (cloud-init and configureIPv6OnVMI),
happen in a different stage of the running, one upon boot, and one in the tests,
does it affect behaviour of NM ?
i mean, would disabling it at configureIPv6OnVMI can be too late compared to disable it in the cloud-init ?

Moreover from what i rememeber, we added the logic configureIPv6OnVMI as a futuristic need,
not because we actually saw a need in a test, i am not sure this is the right way to do so.
Mostly in case the issue above is real.
maybe we need to remove it from there, and to make sure what tests need it,
and for them have it in userData so we need to maintain just one place ?

Reply:
> I think it is unrelated. You're fixing here something that was already done in 'configureIPv6OnVMI' but was missed here. The fix should be the same.
> If we had a way to reuse the same code here and in 'configureIPv6OnVMI' it would be the best. But I don't have an idea how. And if you're going to refactor it anyway, maybe it doesn't worth the effort.

These are not the same, one seems to work on VMI boot (through cloud-init) and the other while it is running.
Therefore, I would not recommend mixing the two.
I do not even understand why there is a need to configure IPv6 in two places (but I do not expect to discuss this here).